### DISCUSSION TITLE
Error "Couldn't open file from storage"

### DISCUSSION THREAD

**Discussion by webdb22** [bitfireAT/icsx5#34] [Q&A] [answered]
Dear experts,

to start with, ICSx5 has worked for me fine for syncing a few calendars to my Android device for quite some time now, just reliable does the job in the background, great!

Unfortunately, since upgrading to V2.0 a few days ago, ICSx5 seems to be unable to read from local storage (which worked fine in V1.9 and previous) using the "file:///..." URL as described in https://icsx5.bitfire.at/usage/

The error message I get:
```
java.io.IOException: Couldn't open file from storage
	at at.bitfire.icsdroid.CalendarFetcher.fetchLocal$icsx5_58_2_0_standardRelease(CalendarFetcher.kt:118)
	at at.bitfire.icsdroid.CalendarFetcher.run(CalendarFetcher.kt:48)
	at at.bitfire.icsdroid.ProcessEventsTask.processEvents(ProcessEventsTask.kt:106)
	at at.bitfire.icsdroid.ProcessEventsTask.run(ProcessEventsTask.kt:39)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at java.lang.Thread.run(Thread.java:764)
```
Software Versions:
- ICSx5 V2.0
- Android V8.0

Not sure if the syntax has changed in V2.0, or additional permissions would be required, or just a bug, in any case any help to fix this would be highly appreciated, thanks!

**Comment by rfc2822** [comment, marked as answer]
Hi,

ICSx⁵ 2.0 now uses the Storage Access Framework (SAF) because `file://` URIs are not supported by Android anymore (for security reasons).

Can you please check whether it works when you remove the subscription and add it again with the Pick file / SAF method?

**Reply by webdb22** [reply]
@rfc2822 Thanks a lot for your quick reply and explanation! 

I've disabled the current subscriptions and added them again using the "Pick File" method, the error message is gone and it seems to work fine again! Thanks a lot for the hint!

Based on that two comments / suggestions:
-  You might want to add that hint to the V2.0 release notes, as you're mentioning the new SAF there, but not the de-support of "old-style" subscriptions, which might not be obvious to everybody (like me ;-) )
- You might want to update https://icsx5.bitfire.at/usage/ to refer to the correct way of subscribing to local files depending on the ICSx5 version

Just my two cents...
